File: python/src/operational/Party.py
from ..knowledge.PartyType import PartyType 

class Party:
    def __init__(self, name: str, party_type: PartyType):
        self.name = name
        self.party_type = party_type
    
    def get_all_types(self):
        return self.party_type.get_all_types()

    def get_type(self):
        return self.party_type

    def __repr__(self):
        return f"Party({self.name}, {self.party_type.name})"

File: src/main/java/org/nc/operational/Party.java
package org.nc.operational;

import org.nc.knowledge.PartyType;

public class Party {
    private final String name;
    private PartyType type;

    public Party(String name, PartyType type) {
        this.name = name;
        this.type = type;
    }

    public String getName() {
        return name;
    }

    public PartyType getType() {
        return type;
    }

    public void setType(PartyType type) {
        this.type = type;
    }
}